### ISSUE TITLE
Create subscriptions for the all entities

### ISSUE THREAD

**Issue by Odrakir** [toggl/ios#681]
## Description
After #680 is merged we should do the same for the rest of entities, the code should be very similar.

Basically creating an observable  in the Repository for every entity, bind it to DB observation and modify the Repository API to throw instead of returning an Observable.

Optionally we could also do the first load of the state via a subscription, but we could also do this further down the road in another issue.

## Dependencies
#680 

## Definition of done
- [ ] Create subscriptions for Project
- [ ] Create subscriptions for Client
- [ ] Create subscriptions for Workspace
- [ ] Create subscriptions for Tag
- [ ] Create subscriptions for Task
